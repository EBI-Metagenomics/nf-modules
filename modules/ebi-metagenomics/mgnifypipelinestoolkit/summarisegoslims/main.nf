
process MGNIFYPIPELINESTOOLKIT_SUMMARISEGOSLIMS {
    tag "$meta.id"
    label 'process_single'

    container 'microbiome-informatics/mgnify-pipelines-toolkit:1.0.4'

    input:
    tuple val(meta), path(interproscan_tsv)
    tuple val(meta2), path(gaf)
    path go_obo
    path go_banding

    output:
    tuple val(meta), path("*_summary.tsv"),      emit: go_summary
    tuple val(meta), path("*_summary_slim.tsv"), emit: goslim_summary
    path "versions.yml",                         emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def args = task.ext.args ?: ''
    def interproscan_tsv_file = interproscan_tsv.name.replace(".gz", "")
    """
    if [ "${interproscan_tsv.name.endsWith(".gz")}" == "true" ]; then
        gzip -c -d $interproscan_tsv > $interproscan_tsv_file
    fi
    summarise_goslims \\
        ${args} \\
        -go ${go_obo} \\
        -gb ${go_banding} \\
        -i ${interproscan_tsv_file} \\
        -gaf ${gaf} \\
        -o ${prefix}_summary

    mv ${prefix}_summary ${prefix}_summary.tsv
    mv ${prefix}_summary_slim ${prefix}_summary_slim.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: \$(get_mpt_version)
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_summary.tsv
    touch ${prefix}_summary_slim.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: \$(get_mpt_version)
    END_VERSIONS
    """
}
