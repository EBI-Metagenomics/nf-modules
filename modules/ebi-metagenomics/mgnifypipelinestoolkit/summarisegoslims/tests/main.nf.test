
nextflow_process {

    name "Test Process MGNIFYPIPELINESTOOLKIT_SUMMARISEGOSLIMS"
    script "../main.nf"
    process "MGNIFYPIPELINESTOOLKIT_SUMMARISEGOSLIMS"

    tag "modules"
    tag "modules_ebimetagenomics"
    tag "mgnifypipelinestoolkit"
    tag "mgnifypipelinestoolkit/summarisegoslims"

    test("summarise gos with interproscan tsv") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${baseDir}/modules/ebi-metagenomics/generategaf/tests/ips_out.tsv", checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${baseDir}/modules/ebi-metagenomics/owltools/tests/ERRTESTING_ips_annotations.gaf", checkIfExists: true)
                ]
                input[2] = file("${baseDir}/modules/ebi-metagenomics/owltools/tests/go-dummy.obo", checkIfExists: true)
                input[3] = file("${moduleDir}/tests/goslim_banding_2024.txt", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("summarise gos with interproscan tsv - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${baseDir}/modules/ebi-metagenomics/generategaf/tests/ips_out.tsv", checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${baseDir}/modules/ebi-metagenomics/owltools/tests/ERRTESTING_ips_annotations.gaf", checkIfExists: true)
                ]
                input[2] = file("${baseDir}/modules/ebi-metagenomics/owltools/tests/go-dummy.obo", checkIfExists: true)
                input[3] = file("${moduleDir}/tests/goslim_banding_2024.txt", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
