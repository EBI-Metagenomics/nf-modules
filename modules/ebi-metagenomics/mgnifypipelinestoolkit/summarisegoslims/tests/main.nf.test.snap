{
    "summarise gos with interproscan tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary.tsv:md5,c24ea2f0b0763052a18190b77a61bd42"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary_slim.tsv:md5,c2ce302a04dd3cfa52488edf0b08744a"
                    ]
                ],
                "2": [
                    "versions.yml:md5,635a55e2cce9bb169a9b4560a8ff0c18"
                ],
                "go_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary.tsv:md5,c24ea2f0b0763052a18190b77a61bd42"
                    ]
                ],
                "goslim_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary_slim.tsv:md5,c2ce302a04dd3cfa52488edf0b08744a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,635a55e2cce9bb169a9b4560a8ff0c18"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-27T22:15:13.348635"
    },
    "summarise gos with interproscan tsv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary_slim.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,635a55e2cce9bb169a9b4560a8ff0c18"
                ],
                "go_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "goslim_summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_summary_slim.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,635a55e2cce9bb169a9b4560a8ff0c18"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-27T22:06:10.801899"
    }
}